contracts/casino/include/casino/casino.hpp

```diff
@@ -145,6 +184,48 @@ class [[eosio::contract("casino")]] casino: public eosio::contract {
 
     void verify_game(uint64_t game_id);
     uint64_t get_game_id(name game_account);
+
+    void session_update(uint64_t ses_id, uint64_t game_id, asset quantity) {
+        session_state_table sessions(_self, _self.value);
+        const auto itr = sessions.find(ses_id);
+        if (itr == sessions.end()) {
+            sessions.emplace(_self, [&](auto& row) {
+                row.ses_id = ses_id;
+                row.game_id = game_id;
+                row.max_win = quantity;
+            });
+        } else {
+            check(itr->game_id == game_id, "invalid game id");
+            sessions.modify(itr, _self, [&](auto& row) {
+                row.max_win += quantity;
+            });
+        }
+        gstate.game_active_sessions_sum += quantity;
+    }
+
+    void session_close(uint64_t ses_id, uint64_t game_id) {
+        session_state_table sessions(_self, _self.value);
+        const auto itr = sessions.require_find(ses_id, "no active game session found");
+        check(itr->game_id == game_id, "invalid game id");
+        gstate.game_active_sessions_sum -= itr->max_win;
+        sessions.erase(itr);
+    }
 };
 
+const asset casino::zero_asset = asset(0, casino::core_symbol);
+
 } // namespace casino
+
+namespace token {
+    struct account
```

Comment:
follow usual code style and put open bracket just after struct name (without newline)